src/server/Server.ts

```diff
@@ -131,12 +132,26 @@ wss.on('connection', (ws, req) => {
                 ? forwarded[0]  // Get the first IP if it's an array
                 : forwarded || req.socket.remoteAddress;
 
+            const username = clientMsg.username;
+            const { isValid, error } = validateUsername(username);
+            if (!isValid) {
+                const errorMsg = error || "Invalid username.";
+                // Send error back to the client
+                ws.send(JSON.stringify({
+                    type: 'error',
+                    input: 'username-input',
+                    message: errorMsg,
+                }));
+                return;
+            }
+
```

Comment:
Instead of sending a error back, I think the flow would be simpler if the server just removed any non alphanumeric characters and then updated the message. Just keeps the code a bit simpler

Reply:
unfortunately spawn messages can update the username too, so we'll need validation for spawn intents :/, but that can be added later.

The reason i allowed users to update their username during spawn is because sometimes users will join a public game, and then change their username afterwards. So this way when they spawn they have another chance to update it.

Reply:
Well that will need more code and i dont like the think of changing username after he enters tha game so i put validation on it and if changes the name we can handle that later i think